python/ray/llm/_internal/serve/configs/server_models.py

```diff
@@ -494,80 +525,50 @@ def get_serve_options(
 
         return deployment_config
 
+    @classmethod
+    def parse_from(
+        cls,
+        args: Union[str, "LLMConfig", Any, Sequence[Union["LLMConfig", str, Any]]],
+    ) -> List[Type["LLMConfig"]]:
+        """Parse the input args and return a standardized list of LLMConfig objects
+
+        Supported args format:
+        1. The path to a yaml file defining your LLMConfig
+        2. The path to a folder containing yaml files, which define your LLMConfigs
+        3. A list of yaml files defining multiple LLMConfigs
+        4. A dict or LLMConfig object
+        5. A list of dicts or LLMConfig objects
+        """
 
-def _is_yaml_file(filename: str) -> bool:
-    yaml_extensions = [".yml", ".yaml", ".json"]
-    for s in yaml_extensions:
-        if filename.endswith(s):
-            return True
-    return False
-
-
-def _parse_path_args(path: str) -> List[LLMConfig]:
-    assert os.path.exists(
-        path
-    ), f"Could not load model from {path}, as it does not exist."
-    if os.path.isfile(path):
-        with open(path, "r") as f:
-            llm_config = LLMConfig.parse_yaml(f)
-            return [llm_config]
-    elif os.path.isdir(path):
-        apps = []
-        for root, _dirs, files in os.walk(path):
-            for p in files:
-                if _is_yaml_file(p):
-                    with open(os.path.join(root, p), "r") as f:
-                        llm_config = LLMConfig.parse_yaml(f)
-                        apps.append(llm_config)
-        return apps
-    else:
-        raise ValueError(
-            f"Could not load model from {path}, as it is not a file or directory."
-        )
-
-
-def parse_args(
-    args: Union[str, LLMConfig, Any, Sequence[Union[LLMConfig, str, Any]]],
-) -> List[LLMConfig]:
-    """Parse the input args and return a standardized list of LLMConfig objects
-
-    Supported args format:
-    1. The path to a yaml file defining your LLMConfig
-    2. The path to a folder containing yaml files, which define your LLMConfigs
-    3. A list of yaml files defining multiple LLMConfigs
-    4. A dict or LLMConfig object
-    5. A list of dicts or LLMConfig objects
-    """
-
-    raw_models = [args]
-    if isinstance(args, list):
-        raw_models = args
-
-    # For each
-    models: List[LLMConfig] = []
-    for raw_model in raw_models:
-        if isinstance(raw_model, str):
-            if os.path.exists(raw_model):
-                parsed_models = _parse_path_args(raw_model)
+        raw_models = [args]
+        if isinstance(args, list):
+            raw_models = args
+
+        # For each
+        models: List[ray.serve.llm.LLMConfig] = []
```

Comment:
do we have to type this list here? I feel weird about using the derived classes for type checking here (tho it's technically right)